sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -66,8 +65,7 @@ private[spark] class HiveExternalCatalog(client: HiveClient) extends ExternalCat
     try {
       body
     } catch {
-      case e: NoSuchItemException =>
-        throw new AnalysisException(e.getMessage)
+      case e: AnalysisException => throw new AnalysisException(e.getMessage)
```

Comment:
The effect of this is to lose the original stack trace and replace it with a new one here. Is that desirable vs just removing this line to let the original exception propagate?